src/Processors/Transforms/FilterTransform.cpp

```diff
@@ -121,6 +127,16 @@ void FilterTransform::doTransform(Chunk & chunk)
 
     if (constant_filter_description.always_true || on_totals)
     {
+        ProfileEvents::increment(ProfileEvents::FilterTransformPassedRows, num_rows_before_filtration);
+
+        size_t num_bytes_before_filtration = 0;
+        for (const auto & column : columns)
+        {
+            if (column)
+                num_bytes_before_filtration += column->byteSize();
+        }
+        ProfileEvents::increment(ProfileEvents::FilterTransformPassedBytes, num_bytes_before_filtration);
+
```

Comment:
Would you mind creating a lambda from this snipped and reuse it? I think the complexity of this few rows reached the point when it makes sense to convert it into a lambda.

Reply:
Done.